src/averageAges.js

```diff
@@ -59,6 +79,19 @@ function calculateAverageAgeDiff(people, onlyWithSon) {
   // 2. keep people who have mothers in the array
   // 3. calculate the difference child.born - mother.born
   // 4. return the average value
+  const peopleWithMothers = onlyWithSon
+    ? people.filter(({ sex, mother }) =>
+      people.find(({ name }) => mother === name) && sex === 'm')
+    : people.filter(({ mother }) =>
+      people.find(({ name }) => mother === name));
+
+  const averageAge = Math.round(peopleWithMothers.reduce(
+    (prev, curr) =>
+      prev + curr.born
+      - people.find(({ name }) => curr.mother === name).born, 0)
+      * 100 / peopleWithMothers.length) / 100;
```

Comment:
Everything is written correctly now, but such code is difficult to read and maintain in the future.
Try to break it into smaller parts.